src/file.ts

```diff
@@ -1536,23 +1536,17 @@ class File extends ServiceObject<File> {
           return;
         }
 
-        // TODO(https://github.com/googleapis/nodejs-storage/issues/709):
-        // Remove once the backend issue is fixed.
-        // If object is stored compressed (having
-        // metadata.contentEncoding === 'gzip') and was served decompressed,
-        // then skip checksum validation because the remote checksum is computed
-        // against the compressed version of the object.
-        if (!isServedCompressed) {
+        // Did the server decompress object before serving it back to us?
+        const serverDecompressed =
+          headers['x-goog-stored-content-encoding'] === 'gzip' && !isCompressed;
+
+        if (serverDecompressed) {
```

Comment:
Shoot, I think you need to remove this code block because it's making a secondary get metadata request. This conditional is meant to go around the validation below.